src/components/PhonesCatalog/PhoneDetailsSingle/PhoneDetailsSingle.js

```diff
@@ -1,57 +1,73 @@
 import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
-import stringSimilarity from 'string-similarity';
 import { Link } from 'react-router-dom';
 import { connect } from 'react-redux';
 
 import { requestImages } from '../../../store/actions/fetchImagesAction';
 import { addPhoneToCart } from '../../../store/actions/inCartAction';
+import { storeCurrentMatch, getSinglePhoneInfo } from '../../../store/actions/fetchSingleItemAction';
+
 import './PhoneDetailsSingle.scss';
 
 const PhoneDetailsSingle = (props) => {
 
-   const { phonesInfo, onImgFetch, addPhoneToCart, match } = props;
+   const { singleItem, fetchSingleItem, isLoading, storeCurrentMatch, onImgFetch, addPhoneToCart, } = props;
    const { images } = props.images;
 
-   const nameToMatch = match.params.brand;
-   const matches = stringSimilarity.findBestMatch(nameToMatch, phonesInfo.map(phoneInfo => phoneInfo.DeviceName).concat(''));
-   const phoneDetails = phonesInfo[matches.bestMatchIndex] || {};
-
-   const phoneDeviceName = phoneDetails.DeviceName;
-   const items = Object.keys(phoneDetails).map((key, index) =>
-      <p key={index}> <strong>{key}</strong> - {phoneDetails[key]}</p>
+   const phoneDeviceName = singleItem.DeviceName;
+   const items = Object.keys(singleItem).map((key, index) =>
+      <p key={index}> <strong>{key}</strong> - {singleItem[key]}</p>
    );
 
+   const matchStoreHandler = () => {
+      const currentMatch = props.match.params.brand.split('_').join(' ');
+      storeCurrentMatch(currentMatch);
+   }
+
    useEffect(() => {
+      matchStoreHandler();
+      fetchSingleItem();
       onImgFetch();
-   }, [onImgFetch]);
+   }, [fetchSingleItem]);
 
    const randomImageSrc = images.length ? images[Math.floor(Math.random() * images.length)] : null;
 
    const addInCartHandle = () => {
-      let currentPhone = props.location.phone;
-      addPhoneToCart([currentPhone]);
+      addPhoneToCart(singleItem);
    }
 
+   const randomPrice = `${Math.floor(Math.random() * 500)} Eur - random`;
+
+   let productPrice = singleItem.price;
+   if (singleItem.price === undefined) {
+      productPrice = randomPrice
+   }
+   if (productPrice.includes(';&thinsp;')) {
+      productPrice = `${productPrice.split('/').slice(0, 1).join().split(';').slice(2, 3)} Eur`;
```

Comment:
Comment `productPrice.split('/').slice(0, 1).join().split(';').slice(2, 3)` with some examples as it is hard to understand output